Mathlib/AlgebraicGeometry/Modules/Tilde.lean

```diff
@@ -252,6 +254,140 @@ noncomputable def localizationToStalk (x : PrimeSpectrum.Top R) :
     (TopCat.Presheaf.stalk (tildeInModuleCat M) x) :=
   LocalizedModule.lift _ (toStalk M x) <| isUnit_toStalk M x
 
+
+/-- The ring homomorphism that takes a section of the structure sheaf of `R` on the open set `U`,
+implemented as a subtype of dependent functions to localizations at prime ideals, and evaluates
+the section on the point corresponding to a given prime ideal. -/
+def openToLocalization (U : Opens (PrimeSpectrum R)) (x : PrimeSpectrum R) (hx : x ∈ U) :
+    (tildeInModuleCat M).1.obj (op U) ⟶
+    ModuleCat.of R (LocalizedModule x.asIdeal.primeCompl M) where
+  toFun s := (s.1 ⟨x, hx⟩ : _)
+  map_add' _ _ := rfl
+  map_smul' _ _ := rfl
+
+/--
+The morphism of `R`-modules from the stalk of `M^~` at `x` to the localization of `M` at the
+prime ideal of `R` corresponding to `x`.
+-/
+noncomputable def stalkToFiberLinearMap (x : PrimeSpectrum.Top R) :
+    TopCat.Presheaf.stalk (tildeInModuleCat M) x ⟶
+    ModuleCat.of R (LocalizedModule x.asIdeal.primeCompl M) :=
+  Limits.colimit.desc ((OpenNhds.inclusion x).op ⋙ (tildeInModuleCat M))
+    { pt := _
+      ι :=
+      { app := fun U => openToLocalization M ((OpenNhds.inclusion _).obj U.unop) x U.unop.2 } }
+
+@[simp]
+theorem germ_comp_stalkToFiberLinearMap (U : Opens (PrimeSpectrum.Top R)) (x) (hx : x ∈ U) :
+    TopCat.Presheaf.germ (tildeInModuleCat M) U x hx ≫ stalkToFiberLinearMap M x =
+    openToLocalization M U x hx :=
+  Limits.colimit.ι_desc _ _
+
+@[simp]
+theorem stalkToFiberLinearMap_germ (U : Opens (PrimeSpectrum.Top R)) (x : PrimeSpectrum.Top R)
+    (hx : x ∈ U) (s : (tildeInModuleCat M).1.obj (op U)) :
+    stalkToFiberLinearMap M x
+      (TopCat.Presheaf.germ (tildeInModuleCat M) U x hx s) = (s.1 ⟨x, hx⟩ : _) :=
+  DFunLike.ext_iff.1 (germ_comp_stalkToFiberLinearMap M U x hx) s
+
+@[simp]
+theorem toOpen_germ (U : Opens (PrimeSpectrum.Top R)) (x) (hx : x ∈ U) :
+    toOpen M U ≫ TopCat.Presheaf.germ (tildeInModuleCat M) U x hx = toStalk M x := by
+  rw [← toOpen_res M ⊤ U (homOfLE le_top : U ⟶ ⊤), Category.assoc, Presheaf.germ_res]; rfl
+
+theorem germ_toOpen (U : Opens (PrimeSpectrum.Top R)) (x) (hx : x ∈ U) (f : M) :
+    (M.tildeInModuleCat.germ U x hx) ((ModuleCat.Tilde.toOpen M U) f) = toStalk M x f := by
```

Comment:
```suggestion
    M.tildeInModuleCat.germ U x hx (ModuleCat.Tilde.toOpen M U f) = toStalk M x f := by
```

Reply:
Does this lemma come for free if you put `@[elementwise]` on `toOpen_germ`?

Reply:
indeed!